### PR TITLE
[Enhance](fe) Support setting initial root password when FE firstly launch

### PR DESCRIPTION
**Pull request by WinkerDu**
## Proposed changes

Issue Number: close #xxx

Doris FE firstly launch with no root password for now. In my organization scene, we use Doris operator start a Doris cluster, then `set password` for root user via  peripheral system, which increase system Complexity.

This pr introduces FE config `initial_root_password` to set root user initial 2-staged SHA-1 encrypted password, default as '', means no root password. Subsequent 'set password' operations for root user will overwrite the initial root password.

if `initial_root_password` is not a valid 2-staged SHA-1 encrypted password, no root password set as before.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by WinkerDu** [comment]
run buildall

**Comment by WinkerDu** [comment]
run buildall

**Comment by WinkerDu** [comment]
cc @morningman @chenlinzhong @luozenglin Please have a review, thx

**Comment by WinkerDu** [comment]
run buildall

**Comment by WinkerDu** [comment]
run buildall

**Comment by WinkerDu** [comment]
run buildall

**Review by luozenglin — DISMISSED** [review]
LGTM

**Comment by luozenglin** [comment]
Maybe documentation should be provided on how to convert a plaintext password to `initial_root_password`.

**Comment by WinkerDu** [comment]
run buildall

**Comment by WinkerDu** [comment]
> Maybe documentation should be provided on how to convert a plaintext password to `initial_root_password`.

@luozenglin new commit is pushed, PTAL

**Review by luozenglin — DISMISSED** [review]
LGTM

**Review by chenlinzhong — DISMISSED** [review]
LGTM

File: docs/en/docs/admin-manual/config/fe-config.md [hunk 1]
```diff
@@ -376,6 +376,18 @@ Is it a configuration item unique to the Master FE node: true
...
 
+#### `initial_root_password`
+
+Set root user initial 2-staged SHA-1 encrypted password, default as '', means no root password. Subsequent 'set password' operations for root user will overwrite the initial root password.
+
+Example: If you want to configure a plaintext password `root@123`. You can execute Doris SQL `select password('root@123')` to generate encrypted password `*A00C34073A26B40AB4307650BFB9309D6BFA6999`.
+
+Default: empty string
+
+s it possible to dynamically configure: false
```
Comment (morningman) [inline docs/en/docs/admin-manual/config/fe-config.md:387]:
```suggestion
Is it possible to dynamically configure: false
```

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java [hunk 2]
```diff
@@ -2293,4 +2293,11 @@ public class Config extends ConfigBase {
                     + "If it is less than this value, it will be diagnosed as balanced."
     })
     public static double diagnose_balance_max_tablet_num_ratio = 1.1;
+
+    @ConfField(masterOnly = true, description = {
+            "设置 root 用户初始化2阶段 SHA-1 加密密码，默认为''，即不设置 root 密码。"
```
[the remark this answers is not in the document]
Comment (WinkerDu) [reply fe/fe-common/src/main/java/org/apache/doris/common/Config.java:2298]: 已加

**Comment by WinkerDu** [comment]
@morningman , docs annotations has optimized, PTAL, thx

**Comment by WinkerDu** [comment]
run buildall

**Comment by WinkerDu** [comment]
run buildall

**Comment by WinkerDu** [comment]
run feut

**Review by luozenglin — APPROVED** [review]
LGTM

**Review by zxealous — APPROVED** [review]
LGTM

**Pull request merged**